src/mapping/mod.rs

```diff
@@ -0,0 +1,157 @@
+use crate::event::Event;
+
+pub mod parser;
+pub mod query;
+
+use string_cache::DefaultAtom as Atom;
+
+pub type Result<T> = std::result::Result<T, String>;
+
+pub trait Function: Send + core::fmt::Debug {
```

Comment:
I've updated the components within the `query` package to use `pub(in crate::mapping)`, as `pub(super)` didn't seem to allow access for sibling packages. I think it would also be reasonable at this point to move `query` components into the parent `mapping` package, but I'll leave it as it is for now as the query codebase is the one most likely to expand the fastest.